File: 0036-valid-sudoku/0036-valid-sudoku.py
class Solution:
  def isValidSudoku(self, board: List[List[str]]) -> bool:
    big = set()
    for i in range(0,9):
      for j in range(0,9):
        if board[i][j]!='.':
          cur = board[i][j]
          if (i,cur) in big or (cur,j) in big or (i//3,j//3,cur) in big:
            return False
          big.add((i,cur))
          big.add((cur,j))
          big.add((i//3,j//3,cur))
    return True

File: 0036-valid-sudoku/0036-valid-sudoku.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        for (int i = 0; i < 9; ++i) {
            int seen[9] = {0};
            for (int j = 0; j < 9; ++j) {
                if (board[i][j] != '.') {
                    if (seen[board[i][j] - '1']) return false;
                    seen[board[i][j] - '1'] = 1;
                }
            }
        }
        for (int i = 0; i < 9; ++i) {
            int seen[9] = {0};
            for (int j = 0; j < 9; ++j) {
                if (board[j][i] != '.') {
                    if (seen[board[j][i] - '1']) return false;
                    seen[board[j][i] - '1'] = 1;
                }
            }
        }
        for (int i = 0; i < 9; i += 3) {
            for (int j = 0; j < 9; j += 3) {
                int seen[9] = {0};
                for (int x = 0; x < 3; ++x) {
                    for (int y = 0; y < 3; ++y) {
                        if (board[i+x][j+y] != '.') {
                            if (seen[board[i+x][j+y] - '1']) return false;
                            seen[board[i+x][j+y] - '1'] = 1;
                        }
                    }
                }
            }
        }
        return true;
    }
};